### PR TITLE
CB-4039 logout with context

### PR DESCRIPTION
**Pull request by alexander-skoblikov**

### CODE REVIEW

File: server/bundles/io.cloudbeaver.service.auth/schema/service.auth.graphqls [hunk 1]
```diff
@@ -139,8 +144,13 @@ extend type Query {
     authUpdateStatus(authId: ID!, linkUser: Boolean): AuthInfo!
 
     # Logouts user. If provider not specified then all authorizations are revoked from session.
+    @deprecated
     authLogout(provider: ID, configuration: ID): Boolean
 
+    # Same as #authLogout, but returns additional information
+    @since(version: "23.3.3")
+    authLogoutExtended(provider: ID, configuration: ID): LogoutInfo!
```
Comment (Wroud) [inline server/bundles/io.cloudbeaver.service.auth/schema/service.auth.graphqls:152]: it's not clear what `Extended` suffix means especially if we will remove `authLogout`

File: webapp/packages/core-authentication/src/UserInfoResource.ts [hunk 2]
```diff
@@ -152,11 +155,25 @@ export class UserInfoResource extends CachedDataResource<UserInfo | null, void,
...
+    if (redirectLinks.length) {
+      const oktaLink = authLogoutExtended?.redirectLinks[0];
+      const id = `okta-logout-id-${getUniqueId()}`;
+
+      this.windowsService.open(id, {
+        url: oktaLink,
+        width: 400,
+        height: 400,
+      });
+    }
```
Comment (Wroud) [inline webapp/packages/core-authentication/src/UserInfoResource.ts]: this logic is out of resource responsibilities

File: webapp/packages/core-utils/src/getUniqueId.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+function getNextIdGenerator() {
+  let id = 0;
+  return () => id++;
+}
+
+const getNextId = getNextIdGenerator();
+
+export function getUniqueId() {
+  return getNextId();
+}
```
Comment (Wroud) [inline webapp/packages/core-utils/src/getUniqueId.ts]: why not to use `uuid`?

File: webapp/packages/core-authentication/src/UserInfoResource.ts (hunk 2 above)
Comment (Wroud) [reply webapp/packages/core-authentication/src/UserInfoResource.ts]: please change `logout` function in `AuthenticationService`

**Review by Wroud — CHANGES_REQUESTED** [review]

File: server/bundles/io.cloudbeaver.service.auth/schema/service.auth.graphqls (hunk 1 above)
Comment (sergeyteleshev) [reply server/bundles/io.cloudbeaver.service.auth/schema/service.auth.graphqls:152]: Alexander decided `authLogout` function to be for a while for back backward compatibility. so this is why we have new function with `Extended` suffix

Comment (Wroud) [reply server/bundles/io.cloudbeaver.service.auth/schema/service.auth.graphqls:152]: problem is that `Extended` means nothing this is not good suffix for api
`authLogoutInfo` sound better

File: webapp/packages/plugin-authentication/src/AuthenticationService.ts [hunk 4]
```diff
@@ -106,7 +107,21 @@ export class AuthenticationService extends Bootstrap {
...
+      if (redirectLinks.length) {
+        const oktaLink = redirectLinks[0];
+        const id = `okta-logout-id-${uuid()}`;
+
+        this.windowsService.open(id, {
+          url: oktaLink,
+          width: 400,
+          height: 400,
+        });
+      }
```
Comment (Wroud) [inline webapp/packages/plugin-authentication/src/AuthenticationService.ts]: this logic must replace 
```ts
if (userAuthConfiguration?.configuration.signOutLink) {
      this.logoutConfiguration(userAuthConfiguration);
    }
```

we need to rewrite or remove it or change and maybe to change `logoutConfiguration`

**Review by Wroud — CHANGES_REQUESTED** [review]

**Pull request merged**